wweval/R/combine_and_summarize_ww_data.R

```diff
@@ -0,0 +1,417 @@
+    dplyr::summarise(
+      n_ww_present = sum(.data$ww_data_present, na.rm = TRUE)
+    )
+
+  n_states_w_complete_ww_data <- ww_metadata |>
+    dplyr::group_by(.data$location) |>
+    dplyr::summarize(
+      complete_ww = all(.data$ww_data_present == 1)
+    ) |>
+    ungroup() |>
+    dplyr::summarize(n_complete_ww = sum(.data$complete_ww)) |>
+    dplyr::pull(.data$n_complete_ww)
+
+  n_states_w_no_ww_data <- ww_metadata |>
+    dplyr::group_by(.data$location) |>
+    dplyr::summarize(
+      no_ww = all(.data$ww_data_present == 0)
+    ) |>
+    dplyr::ungroup() |>
+    dplyr::summarize(n_zero_ww = sum(.data$no_ww)) |>
+    dplyr::pull(.data$n_zero_ww)
+
+  n_combos_w_hosp_conv_flags <- ww_metadata |>
+    dplyr::summarise(n_hosp_flags = sum(.data$any_flags_hosp, na.rm = TRUE)) |>
+    dplyr::pull(.data$n_hosp_flags)
+  n_combos_w_ww_conv_flags <- ww_metadata |>
+    dplyr::summarise(n_ww_flags = sum(.data$any_flags_ww, na.rm = TRUE)) |>
+    dplyr::pull(.data$n_ww_flags)
+
+  n_insuff_ww <- ww_metadata |>
+    dplyr::summarise(
+      n_ww_insuff = sum(.data$ww_sufficient == FALSE, na.rm = TRUE)
+    ) |>
+    dplyr::pull(n_ww_insuff)
+
+  n_ww_excluded <- ww_metadata |>
+    dplyr::summarise(
+      n_ww_excl = sum(.data$ww_exclude_manual, na.rm = TRUE)
+    ) |>
+    dplyr::pull(.data$n_ww_excl)
+
+  n_w_ww_expected <- ww_metadata |>
+    dplyr::mutate(
+      ww_expected = dplyr::case_when(
+        ww_data_present == 1 & is.na(.data$ww_exclude_manual) &
+          !isTRUE(.data$any_flags_hosp) & !isTRUE(.data$any_flags_ww) &
+          isTRUE(.data$ww_sufficient) ~ TRUE,
+        TRUE ~ FALSE
+      )
+    ) |>
+    dplyr::summarise(
+      n_expected_exclude = sum(.data$ww_expected, na.rm = TRUE)
+    ) |>
+    dplyr::pull(.data$n_expected_exclude)
+
+  n_no_ww_expected <- nrow(ww_metadata) - n_w_ww_expected
+
+  summary_table <- tibble::tibble(
+    n_forecast_date_states = n_combinations,
+    n_forecast_date_states_w_ww = n_combos_w_ww_data,
+    prop_combos_w_ww = n_combos_w_ww_data / n_combinations,
+    n_states_w_complete_ww_data,
+    n_states_w_no_ww_data,
+    n_combos_w_hosp_conv_flags,
+    n_combos_w_ww_conv_flags,
+    n_insuff_ww,
+    n_ww_excluded,
+    n_no_ww_expected,
+    n_no_ww_actual,
+    n_w_ww_expected,
+    n_w_ww_actual
+  )
+
+
+  # Summarize across states by forecast date
+  forecast_date_summary_table <-
+    ww_metadata |>
+    dplyr::group_by(forecast_date) |>
+    dplyr::summarize(
+      prop_states_w_ww = sum(ww_data_present) / dplyr::n(),
+      pop_coverage_by_date = sum(pop_coverage * state_pop, na.rm = TRUE) / sum(state_pop),
+      avg_avg_latency = mean(avg_latency, na.rm = TRUE),
+      avg_avg_sampling_freq = mean(avg_sampling_freq, na.rm = TRUE),
+      n_states_w_duplicate_obs = sum(n_duplicate_obs > 0, na.rm = TRUE)
+    )
+
+  # Summarize across forecast dates by state
+  state_summary_table <-
+    ww_metadata |>
+    dplyr::group_by(location) |>
```

Comment:
```suggestion
    dplyr::group_by("location") |>
```

Reply:
This didn't work, this needs to be `.data$location`. Same with below.  So `group_by()` is also not tidy selection but tidy evaluation  it seems